src/EventSubscriber/PartySubscriber.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace App\EventSubscriber;
+
+use App\Entity\Party;
+use App\Repository\CasePartyRelationRepository;
+use App\Service\PartyHelper;
+use Doctrine\Bundle\DoctrineBundle\EventSubscriber\EventSubscriberInterface;
+use Doctrine\ORM\Events;
+use Doctrine\Persistence\Event\LifecycleEventArgs;
+
+class PartySubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var CasePartyRelationRepository
+     */
+    private $relationRepository;
+    /**
+     * @var PartyHelper
+     */
+    private $partyHelper;
+
+    public function __construct(CasePartyRelationRepository $relationRepository, PartyHelper $partyHelper)
+    {
+        $this->relationRepository = $relationRepository;
+        $this->partyHelper = $partyHelper;
+    }
+
+    public function preUpdate(LifecycleEventArgs $args)
+    {
+        $object = $args->getObject();
+        $changeSet = $args->getEntityChangeSet();
+        if ($object instanceof Party && array_key_exists('name', $changeSet)) {
```

Comment:
Adding a comment on what is going on here might be useful.

Reply:
Added comment to explain whats happening.